server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -769,17 +779,22 @@ private TSExecuteStatementResp internalExecuteQueryStatement(String statement,
         }
       }
 
+      // remove query info in QueryTimeManager
+      QueryTimeManager.getInstance().unRegisterQuery(queryId);
```

Comment:
Query may not finish in one rpc call, user may call two or more rpc to fetch complete result set. So, you shouldn't remove it after the first call finished.